src/store/wallet/wallet-epics.ts

```diff
@@ -54,7 +55,26 @@ const loadTokenAssetsEpic = (action$: Observable<Action>) =>
           }
         )
       ).pipe(
-        map(({ data }) => loadTokenBalancesActions.success(data.balances)),
+        switchMap(({ data }) =>
+          forkJoin(
+            data.balances.map(async balance => {
+              try {
+                const bestTokenMetadata = await getTokenMetadata(balance.contract, balance.token_id);
+
+                return {
+                  ...balance,
+                  decimals: bestTokenMetadata.decimals,
+                  symbol: bestTokenMetadata.symbol,
+                  name: bestTokenMetadata.name,
+                  thumbnail_uri: bestTokenMetadata.thumbnailUri
+                };
+              } catch (e) {
+                return balance;
+              }
+            })
+          )
+        ),
+        map(balances => loadTokenBalancesActions.success(balances)),
```

Comment:
the same as here https://github.com/madfish-solutions/templewallet-mobile/pull/127#discussion_r674558138

Reply:
(the first part about `async/await`)